src/dos/cdrom_image.cpp

```diff
@@ -700,14 +834,19 @@ bool CDROM_Interface_Image::LoadIsoFile(char* filename)
 
 bool CDROM_Interface_Image::CanReadPVD(TrackFile *file, int sectorSize, bool mode2)
 {
-	Bit8u pvd[COOKED_SECTOR_SIZE];
+	// Guard: Bail if our file pointer is empty
+	if (file == nullptr) return false;
+
+	// Initialize our array in the event file->read() doesn't fully write it
+	Bit8u pvd[COOKED_SECTOR_SIZE] = {0};
+
 	int seek = 16 * sectorSize;	// first vd is located at sector 16
 	if (sectorSize == RAW_SECTOR_SIZE && !mode2) seek += 16;
 	if (mode2) seek += 24;
 	file->read(pvd, seek, COOKED_SECTOR_SIZE);
 	// pvd[0] = descriptor type, pvd[1..5] = standard identifier, pvd[6] = iso version (+8 for High Sierra)
-	return ((pvd[0] == 1 && !strncmp((char*)(&pvd[1]), "CD001", 5) && pvd[6] == 1) ||
-			(pvd[8] == 1 && !strncmp((char*)(&pvd[9]), "CDROM", 5) && pvd[14] == 1));
+	return ((pvd[0] == 1 && !strncmp((char*)(&pvd[1]), "CD001", 5) && pvd[6]  == 1) ||
+	        (pvd[8] == 1 && !strncmp((char*)(&pvd[9]), "CDROM", 5) && pvd[14] == 1));
```

Comment:
Maybe just compare to strings `"\1CD001\1"` and `"\1CDROM\1"`?

Reply:
Legacy code that I was just aligning to read cleaner; reluctant to touch its actual content.